main.tf

```diff
@@ -11,6 +11,30 @@ resource "aws_vpc" "this" {
 }
 
 ###################
+# DHCP Options Set
+###################
+resource "aws_vpc_dhcp_options" "this" {
+  count = "${var.enable_dhcp_options ? 1 : 0}"
+
+  domain_name          = "${var.dhcp_options_domain_name}"
+  domain_name_servers  = "${var.dhcp_options_domain_name_servers}"
+  ntp_servers          = "${var.dhcp_options_ntp_servers}"
+  netbios_name_servers = "${var.dhcp_options_netbios_name_servers}"
+  netbios_node_type    = "${var.dhcp_options_netbios_node_type}"
+}
+
+###############################
+# DHCP Options Set Association
+###############################
+resource "aws_vpc_dhcp_options_association" "this" {
+  count = "${var.enable_dhcp_options ? 1 : 0}"
+
+  vpc_id          = "${aws_vpc.this.id}"
+  dhcp_options_id = "${aws_vpc_dhcp_options.this.id}"
+  depends_on      = ["aws_vpc.this", "aws_vpc_dhcp_options.this"]
```

Comment:
`depends_on` is not really needed here. Or, if this is this an edge case then provide an open issue link.

Reply:
Agreed - no edge case that I know of, I'm not entirely sure why I put that in there.  Fixed!